strawberryfields/api/result.py

```diff
@@ -107,6 +103,70 @@ def state(self):
             raise AttributeError("The state is undefined for a stateless computation.")
         return self._state
 
+    @staticmethod
+    def combine_samples(samples_list, mode_order):
+        """Combine samples measured at different times into one nested array.
+
+        Args:
+            samples_list (list[array, tensor]): the sample measurements in a list
+            mode_order (list[array]): The nested mode order for the measured modes. Must have the
+                same structure as samples_list.
+
+        Returns:
+            array: the samples in ascending mode order with shape ``(shots, measured_modes)``
+        """
+
+        shots = len(samples_list[0])
+
+        mode_order_flat = np.hstack(mode_order)
+        len_modes = len(set(mode_order_flat))
```

Comment:
(Minor suggestion): `np.unique` seems to perform better 
```suggestion
        len_modes = len(np.unique(mode_order_flat))
```
```python
nums = np.repeat(np.array([1, 2, 3]), 10000)
%timeit len(np.unique(nums))
%timeit len(set(nums))
```
outputs
```
248 µs ± 11.2 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
2.47 ms ± 14.1 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

Reply:
Nice. I'll change that!